packages/utils/src/Pending.ts

```diff
@@ -34,7 +44,7 @@ class Pending {
    * Remove a promise from tracking.
    * @param {Promise} promise Promise to stop tracking
    */
-  remove(promise) {
+  remove(promise: Promise<unknown>): void {
```

Comment:
Should this be `CancelablePromise<unknown>`? I suppose Promise works just fine, but if we are converting to CancelablePromises always when adding (and the tracking array is only cancelable), then this should be Cancelable I think.

Since it's checking the promise is the same as the one to remove, passing it the non-cancelable version of the promise would mean it's not actually removed I think